core/base/src/TApplication.cxx

```diff
@@ -628,21 +630,416 @@ TApplication::EExitOnException TApplication::ExitOnException(TApplication::EExit
+      Printf("   pwd                 : show current directory, pad and style");
+      Printf("   ls                  : list contents of current directory");
+      Printf("   which [file]        : shows path of macro file");
+      Printf("   .help Class         : opens the reference gude for that class");
+      Printf("   .help Class::Member : opens the reference gude for function/member");
+      return;
+   }
+
+   // If the user wants to use the extended ".help className" command to access
+   // the online reference guide.
+   if (!strippedCommand.BeginsWith(".help ")) {
+      if (!strippedCommand.BeginsWith(".? ")) {
+         Error("Help", "Unknown command!");
+         return;
+      }
+   }
+   // We remove the command ".help" or ".?" from the TString.
+   if (strippedCommand.BeginsWith(".? ")){
+      strippedCommand.Remove(0, 3);
+   } else {
+      strippedCommand.Remove(0, 5);
+   }
+   // We strip the TString and substract the name of the method.
+   TString strippedClass = strippedCommand.Strip(TString::kBoth);
+   TString stripped = strippedClass;
+   TString memberName = TClassEdit::GetUnqualifiedName(strippedClass.Data());
+   TString className;
+   // If after stripping the string is the same, user is searching for a class/namespace.
+   if (strippedClass == memberName){
+      className = memberName;
+   } else {
+      // Otherwise we already have the name of the method and extract the name of the class.
+      className = strippedClass.Remove(strippedClass.Length() - memberName.Length() - 2);
+   }
+
+   // We check if "className" exists.
+   TClass *cl = TClass::GetClass(className);
+   if (!cl) {
+      // If the user wants to open a browser with the URL for "className" only, but types the class incorrectly.
+      Warning("TApplication::Help", "\"%s\" does not exist in ROOT!", className.Data());
+      return;
+   }
+   // We have enumerators for the two available cases - class and namespace.
+   // We check what is the used mode and define "scopeType".
+   EUrl scopeType;
+   if (cl->Property() & kIsNamespace) {
+      scopeType = kURLforNameSpace;
+   } else {
+      scopeType = kURLforClass;
+   }
+   // We check if the user doesn't want to search for a certain method.
+   if (stripped == memberName){
+      // If so, we open the browser with the URL for the class/namespace
+      // with respect to the "scopeType".
+      OpenInBrowser(UrlGenerator(className, scopeType));
+      return;
+   }
+   // If the user wants to search for a method, we take its name (memberName) and
+   // modify if - we delete everything after the first "(" so the user won't have to
+   // do it by hand when he uses Tab.
+   int bracket = memberName.First("(");
+   if (bracket > 0) {
+      memberName.Remove(bracket,(memberName.Length()-bracket));
+   }
+   // We check if "memberName" is a member fuction of "cl" or any of its base classes.
+   if (TFunction *func = (TFunction *)(cl->GetMethodAllAny(memberName))) {
```

Comment:
`GetMethodAllAny` already returns a `TMethod*` - no need to cast to a `TFunction*`.